src/main/scala/ch/openolitor/stammdaten/KorbHandler.scala

```diff
@@ -308,22 +308,52 @@ trait KorbHandler extends KorbStatusHandler
     }
   }
 
-  def modifyKoerbeForAboVertriebChange(abo: Abo, orig: Option[Abo])(implicit personId: PersonId, session: DBSession, publisher: EventPublisher): Unit = {
+  def modifyKoerbeForAboVertriebChange(abo: HauptAbo, orig: Option[HauptAbo])(implicit personId: PersonId, session: DBSession, publisher: EventPublisher): Unit = {
     logger.debug(s"modifyKoerbeForAboVertriebChange abo: $abo orig: $orig")
     for {
       originalAbo <- orig
       if (abo.vertriebId != originalAbo.vertriebId)
       abotyp <- stammdatenWriteRepository.getAbotypById(abo.abotypId)
     } yield {
+      /* create or delete basket for the main abo*/
+      val offenLieferungen = stammdatenWriteRepository.getLieferungenOffenByAbotyp(abo.abotypId)
       stammdatenWriteRepository.getLieferungenOffenByAbotyp(abo.abotypId) map { lieferung =>
         if (lieferung.vertriebId == originalAbo.vertriebId) {
           deleteKorb(lieferung, originalAbo)
         }
         if (lieferung.vertriebId == abo.vertriebId) {
           upsertKorb(lieferung, abo, abotyp)
         }
-        recalculateNumbersLieferung(lieferung)
       }
+
+      /* create or delete basket for the zusatzabo*/
+      stammdatenWriteRepository.getZusatzAbos(abo.id).filter(z => z.aktiv) map { zusatzabo =>
+        val offenLieferungenForMainAbo = stammdatenWriteRepository.getLieferungenOffenByAbotyp(abo.abotypId)
```

Comment:
Is offenLieferungenForMainAbo not always the same? You could do that outside the loop

Reply:
You're right. I've changed it